src/overlays/actors/ovl_En_Po_Fusen/z_en_po_fusen.c

```diff
@@ -21,34 +29,232 @@ const ActorInit En_Po_Fusen_InitVars = {
+    if ((this->actor.params & 0x8000) != 0) {
+        EnPoFusen_InitFuse(this);
+        return;
+    }
+    EnPoFusen_Init2(this);
+}
+
+void EnPoFusen_Destroy(Actor* thisx, GlobalContext *gCtx) {
+    EnPoFusen* this = THIS;
+    Collision_FiniSphere(gCtx, &this->collider);
+}
+
+u16 EnPoFusen_CheckParent(EnPoFusen *this, GlobalContext *globalCtx) {
+    struct Actor *actorPtr;
+
+    actorPtr = globalCtx->actorCtx.actorList[4].first;
+    if ((this->actor.params & 0x8000) != 0) {
+        return 1;
+    }
+    if (actorPtr != 0) {
+        do{
+            if (actorPtr->id == ACTOR_EN_MA4) {
+                this->actor.parent = actorPtr;
+                return 1;
+            }
+            actorPtr = actorPtr->next;
+        } while (actorPtr != 0);
+    }
+    return 0;
+}
+
+u16 EnPoFusen_CheckCollision(EnPoFusen *this, GlobalContext *gCtx) {
+    
+    if ((u32) this->actionFunc == (u32) EnPoFusen_IdleFuse ) {
+      return 0;
+    }
+
+    this->collider.params.colInfo.loc.x = (s16) this->actor.world.pos.x;
+    this->collider.params.colInfo.loc.y = (s16) (this->actor.world.pos.y + 20.0f);
+    this->collider.params.colInfo.loc.z = (s16) this->actor.world.pos.z;
+    if (((this->collider.base.flagsAC & 2) != 0) 
+      && (this->actor.colChkInfo.damageEffect == 0xF)) {
+        this->collider.base.flagsAC &= ~0x2;
+        return 1;
+    }
+    Collision_AddOT(gCtx, &gCtx->colCheckCtx, &this->collider);
+    Collision_AddAC(gCtx, &gCtx->colCheckCtx, &this->collider);
+    return 0;
+}
+
+void EnPoFusen_Init2(EnPoFusen *this) {
+    this->actor.shape.rot.z = 0;
+    this->actionFunc = EnPoFusen_Idle;
+}
+
+void EnPoFusen_Idle(EnPoFusen *this, GlobalContext *gCtx) {
+    f32 shadowScaleTmp;
+    f32 shadowAlphaTmp;
+    f32 heightOffset;
+    f32 f255 = 255.0f;
+
+    this->actor.world.pos = this->actor.home.pos;
+    this->randScaleChange += 400;
+    this->randXZRotChange += 1500;
+    this->randBaseRotChange += 2500;
+    heightOffset = Math_Sins(this->randScaleChange);
+    heightOffset = 50.0f * heightOffset;
+    this->actor.shape.rot.y += this->randYRotChange;
+    this->actor.world.pos.y += heightOffset;
+    this->actor.shape.rot.z = (s16) (Math_Sins(this->randBaseRotChange) * 910.0f);
+    if ((this->randScaleChange < 0x4000) && (this->randScaleChange >= -0x3FFF)) {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 910, 20, 3000, 100);
+    } else {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 1820, 8, 3000, 100);
+    }
+    this->avgBaseRotation = this->limb9Rot * 3;
+    this->limb3Rot = (u16) (s32) (Math_Sins(this->randBaseRotChange + 0x38E3) * (f32) this->avgBaseRotation);
+    this->limb46Rot = (u16) (s32) (Math_Sins(this->randBaseRotChange)          * (f32) this->avgBaseRotation);
+    this->limb57Rot = (u16) (s32) (Math_Sins(this->randBaseRotChange - 0x38E3) * (f32) this->avgBaseRotation);
+    this->limb8Rot = (u16) (s32) (Math_Sins(this->randBaseRotChange - 0x71C6) * (f32) this->avgBaseRotation);
```

Comment:
Are these casts needed?

Reply:
Could have sworn I tried to remove them and it broke something, but I guess that was just the (s8) below, can remove after all.